lib/core/text/u16_string.nit

```diff
@@ -0,0 +1,125 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Wrapper of ICU's UTF-16 encoded strings and conversion
+# This module is meant to ease the use of complex string operations provided by the ICU library.
+# The module provides a wrapper for ICU's string structure : `UChar *` as well as conversion fucntions to/from `String` and `CString`
+module u16_string is pkgconfig ("icu-io", "icu-i18n", "icu-uc")
+
+`{
+	#include <unicode/utypes.h>
+	#include <unicode/ustring.h>
+`}
+
+# UTF-16 encoded string
+class U16String
+	super Finalizable
+
+	# Pointer to a `UChar *` string
+	var uchar_string : UCharString is noinit
+
+	# Number of characters in `uchar_string`
+	var length = 0
+
+	# Number of bytes in `uchar_string`
+	var bytes = 0
+
+	# Returns a converted `UCharString` from a `String`
+	init (source : String)
+	do
```

Comment:
Most of your `do`s are inlined with the method definition in the file below.

Reply:
Does it mean that all `do`s should be inline or on their own line throughout the file ?

Reply:
I don't think there's a clear winner in terms of where to write the `do`, I like to inline it, @privat prefers to write it on the following line.

Basically do whichever you prefer, but please keep style consistent within the same source code file

Reply:
don't care. just be consistent